### PR TITLE
 :hammer: Refactor chart-sync

### PR DESCRIPTION
**Pull request by Marigold**
This PR addresses a couple of problems. @lucasrodes if you find it unreadable, I can split it into multiple PRs.

## Fix chart-leakage from data & metadata
See https://github.com/owid/etl/issues/2880. I split function `modified_charts_by_admin` into two - one for chart configs and the other for data & metadata updates. It feels much more intuitive now. It was just rubbish before... so I recommend to take a fresh look at the new logic (check out the docstring for more details). It's not perfect, but a step forward.

## Refactor chart-sync
* Use `ChartDiffsLoader` in `chart-sync`
* Remove `ChartDiff.from_chart_id` method (making `modified_charts_by_admin` the main method)
* Remove revisions from `chart-sync` (make `--chartdiff` default)

## TODO after merging
- [ ] in ops, remove `--chartdiff` flag from `chart-sync`

### CODE REVIEW

File: apps/wizard/app_pages/chart_diff/chart_diff_show.py [hunk 1]
```diff
@@ -157,6 +157,7 @@ def _push_status_binary(self, session: Optional[Session] = None) -> None:
 
     def _pull_latest_chart(self):
         """Get latest chart version from database."""
+        # TODO: can we use ChartDiffLoader for this and get rid of `from_chart_id`?
```
Comment (lucasrodes) [inline apps/wizard/app_pages/chart_diff/chart_diff_show.py:160]: Yeah, totally. It'd be also nice to merge some of the logic from `ChartDiff.from_chart_id` with `ChartDiff.from_charts_df`

File: etl/grapher_model.py [hunk 2]
```diff
@@ -689,88 +689,6 @@ def load_sources(
...
-                ).all()
-            )
-
-    @classmethod
-    def load_revisions(cls, session: Session, chart_id: int) -> List["SuggestedChartRevisions"]:
-        return list(
-            session.scalars(select(SuggestedChartRevisions).where(SuggestedChartRevisions.chartId == chart_id)).all()
-        )
-
-
```
Comment (lucasrodes) [inline etl/grapher_model.py:773]: 🔪🩸

File: apps/wizard/app_pages/chart_diff/chart_diff_show.py (hunk 1 above)
Comment (Marigold) [reply apps/wizard/app_pages/chart_diff/chart_diff_show.py]: Done!

File: apps/wizard/app_pages/chart_diff/chart_diff.py [hunk 3]
```diff
@@ -473,80 +415,133 @@ def get_diffs_summary_df(self, cache: bool = False, **kwargs) -> pd.DataFrame:
...
+    )
+
+    source_df = source_df.set_index(["chartId", "catalogPath"])
+    target_df = target_df.set_index(["chartId", "catalogPath"])
+
+    # align dataframes with INNER join
+    # the inner join is on purpose, because we only want to compare variables that are used in both environments
+    # if the variable is not present in target, it could mean that the chart was updated in target (but we don't
+    # care about that, because it's not a data/metadata change, but chart config change)
+    source_df, target_df = source_df.align(target_df, join="inner")
```
Comment (lucasrodes) [inline apps/wizard/app_pages/chart_diff/chart_diff.py:494]: good move!

**Review by lucasrodes — APPROVED** [review]
lgtm, let's try this!

**Pull request merged**